packages/toolkit/src/dynamicMiddleware/index.ts

```diff
@@ -1,27 +1,29 @@
-import type {
-  Middleware,
-  Dispatch as ReduxDispatch,
-  UnknownAction,
-} from 'redux'
+import type { Dispatch, Middleware, UnknownAction } from 'redux'
 import { compose } from 'redux'
 import { createAction } from '../createAction'
 import { isAllOf } from '../matchers'
 import { nanoid } from '../nanoid'
 import { emplace, find } from '../utils'
 import type {
-  WithMiddleware,
   AddMiddleware,
-  MiddlewareEntry,
   DynamicMiddleware,
   DynamicMiddlewareInstance,
+  MiddlewareEntry,
+  WithMiddleware,
+} from './types'
+export type {
+  DynamicMiddlewareInstance,
+  GetDispatch,
+  GetState,
+  MiddlewareApiConfig,
 } from './types'
 
 const createMiddlewareEntry = <
   State = any,
-  Dispatch extends ReduxDispatch<UnknownAction> = ReduxDispatch<UnknownAction>,
+  DispatchType extends Dispatch<UnknownAction> = Dispatch<UnknownAction>,
```

Comment:
:question: This seems cosmetic - why was this needed?

Reply:
The renames were done because `tsup` sometimes has a hard time with aliasing type imports so something like:

```ts
import type { Dispatch as ReduxDispatch } from 'redux'
```

It somehow renames `ReduxDispatch` back to `Dispatch` which causes issues when you have a generic type parameter with the same name:

```ts
export type AddMiddleware<
  State = any,
  Dispatch extends ReduxDispatch<UnknownAction> = ReduxDispatch<UnknownAction>,
> = {
  (...middlewares: Middleware<any, State, Dispatch>[]): void
  withTypes<MiddlewareConfig extends MiddlewareApiConfig>(): AddMiddleware<
    GetState<MiddlewareConfig>,
    GetDispatch<MiddlewareConfig>
  >
}
```

which can turn into:

```ts
export type AddMiddleware<
  State = any,
  Dispatch extends Dispatch<UnknownAction> = Dispatch<UnknownAction>,
> = {
  (...middlewares: Middleware<any, State, Dispatch>[]): void
  withTypes<MiddlewareConfig extends MiddlewareApiConfig>(): AddMiddleware<
    GetState<MiddlewareConfig>,
    GetDispatch<MiddlewareConfig>
  >
}
```

Which can cause issues.